pkg/contexts/ocm/transfer/autohandler.go

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package transfer
+
+import (
+	"sync"
+
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/transfer/transferhandler"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/transfer/transferhandler/standard"
+	"github.com/open-component-model/ocm/pkg/errors"
+)
+
+func init() {
+	SetDefaultTransferHandlerFactory(func() transferhandler.TransferHandler { h, _ := standard.New(); return h })
+}
+
+var (
+	lock           sync.Mutex
+	defaultHandler func() transferhandler.TransferHandler
+)
+
+func SetDefaultTransferHandlerFactory(f func() transferhandler.TransferHandler) {
+	lock.Lock()
+	defer lock.Unlock()
+	defaultHandler = f
+}
+
+func NewDefaultTransferHandler() transferhandler.TransferHandler {
+	lock.Lock()
+	defer lock.Unlock()
+	return defaultHandler()
+}
+
+// NewTransferHandler creates a transfer handler for the given set of transfer
+// options. If there is no handler type supporting all the given options
+// an ErrNotSupported error is returned.
+// If no handler can be found according to the origin handlers of given options
+// (an option always belongs to a dedicated option set of a dedicated handler
+// but may be supported by other option sets, also) the options sets of all
+// registered handler types with checked by trying the most significant
+// handlers, first, to find the best matching handler for the given option set.
+func NewTransferHandler(list ...transferhandler.TransferOption) (h transferhandler.TransferHandler, ferr error) {
+	var opts transferhandler.TransferOptions
+
+	created := -1
+outer:
+	for {
+		ferr = nil
+		for i := 0; i < len(list); i++ {
+			o := list[i]
+			if o == nil {
```

Comment:
This is interesting. How would this be possible? Why would someone pass in a nil option? Or would the option somehow evaluate to nil? This seems to be a bug then.

Reply:
That is a typical case of defensive programming. Depending on the composition of the options by the caller it should be possible to handle a nil option, also. This will avoid unnecessary Panics and makes the live easier for the caller.

For example, he composes the options by a set of optional command line options, which will be provided in this library typically by option objects, which can be obtained from an option list of a command with <code>optionpackage.From(list)</code>.
This will return nil, if the option is not configured for the actually handled command.